### PR TITLE
BAC-296: update `Client.Evaluate` call to version 1.2alpha1 (breaking)

### PR DESCRIPTION
**Pull request by magnusgohn**
i have updated the clarify go sdk to support the new `evaluate` parameter `groups`. i have also updated some names to match the changes introduced in BAC-282. lastly i have updated the `automation` and `examples` packages to use the new builder style `EvaluateRequest` type

### CODE REVIEW

File: automation/doc.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022-2023 Searis AS
+// Copyright 2022-2024 Searis AS
```
Comment (smyrman) [inline automation/doc.go:1]: You only need to update the copy-right header for files you have changed; nice if you can revert this change for files that are otherwise unchanged in the commit.

File: fields/evaluate.go [hunk 2]
```diff
@@ -21,113 +21,191 @@ import (
...
+const (
+	GroupAggregateDefault GroupAggregateMethod = iota
+	GroupAggregateCount
+	GroupAggregateMin
+	GroupAggregateMax
+	GroupAggregateSum
+	GroupAggregateAvg
+)
+
+type GroupAggregateMethod uint8
```
Comment (smyrman) [inline fields/evaluate.go:46]: Since we go with "aggregation" in the filed name, lets rename to `TimeAggreation...` and `GroupAggregation...`.

for the aggregation method type, let's drop the `Method` suffix  and go with
```
type TimeAggregation uint8
type GroupAggregation uint8
```

Otherwise, remove `StateHist` from the `TimeAggregation` names

File: fields/evaluate.go [hunk 3]
```diff
@@ -21,113 +21,191 @@ import (
...
+	case GroupAggregateMin:
+		return []byte("min"), nil
+	case GroupAggregateMax:
+		return []byte("max"), nil
+	case GroupAggregateSum:
+		return []byte("sum"), nil
+	case GroupAggregateAvg:
+		return []byte("avg"), nil
+	}
+	return nil, fmt.Errorf("unknown aggregation method")
```
Comment (smyrman) [inline fields/evaluate.go:126]: use "bad aggregation method" for consistency with the Unmarshal function.

File: fields/evaluate.go
Comment (smyrman) [inline fields/evaluate.go:1]: There are some formatting issues with these files; ensure that you have the right tooling installed. Preferably set to use `gofmpt` over `go fmt`.

LanguageServer / Vs Code extensions should format the files correctly. Alternatively, you can use the `gofmpt -w` (strict alternative to `go fmt -w`) command line tool to format the files on the terminal.

**Review by smyrman — COMMENTED** [review]
Ensure you enable an automatic formatter.

File: automation/doc.go (hunk 1 above)
Comment (magnusgohn) [reply automation/doc.go:1]: i did this due to a linter error i encountered, so i figured i'd fix all the discrepancies

File: fields/evaluate.go (hunk 2 above)
Comment (magnusgohn) [reply fields/evaluate.go:46]: can do!

**Comment by magnusgohn** [comment]
thank you sindre, i'll be back with something corrected

File: fields/evaluate.go [hunk 4]
```diff
@@ -21,113 +21,190 @@ import (
...
 		return []byte("avg"), nil
-	case AggregateStateHistSeconds:
+	case TimeAggregationSeconds:
 		return []byte("state-histogram-seconds"), nil
-	case AggregateStateHistPercent:
+	case TimeAggregationPercent:
 		return []byte("state-histogram-percent"), nil
-	case AggregateStateHistRate:
+	case TimeAggregationRate:
 		return []byte("state-histogram-rate"), nil
```
Comment (smyrman) [inline fields/evaluate.go:80]: Update to remove `-histogram` in result. The old values work, but the new ones are more canonical.

File: fields/evaluate.go [hunk 5]
```diff
@@ -21,113 +21,190 @@ import (
...
+		*m = TimeAggregationAvg
 	case "state-histogram-seconds":
-		*m = AggregateStateHistSeconds
+		*m = TimeAggregationSeconds
 	case "state-histogram-percent":
-		*m = AggregateStateHistPercent
+		*m = TimeAggregationPercent
 	case "state-histogram-rate":
-		*m = AggregateStateHistRate
+		*m = TimeAggregationRate
```
Comment (smyrman) [inline fields/evaluate.go:104]: Update here as well:

Use:
```
case "state-seconds", "state-histogram-seconds":
```
To handle both the old and the new aggregation names

File: fields/evaluate.go [hunk 6]
```diff
@@ -21,113 +21,190 @@ import (
...
+		*m = GroupAggregationSum
+	case "avg":
+		*m = GroupAggregationAvg
+	default:
+		return fmt.Errorf("bad aggregation method")
+	}
+	return nil
+}
+
+type Item struct {
```
Comment (smyrman) [inline fields/evaluate.go:149]: We can not name this filed simply "Item", as it can be confused to mean a Clarify Item (resource). Instead, this is referring specifically to the _items_ field on the evaluate method.

We need to call it something else. `EvaluateItem` could work better than `ItemAggregation`.

```suggestion
type EvaluateItem struct {
```

File: fields/evaluate.go [hunk 7]
```diff
@@ -21,113 +21,190 @@ import (
...
+	Alias           string                `json:"alias,omitempty"`
+	ID              string                `json:"id,omitempty"`
+	TimeAggregation TimeAggregationMethod `json:"timeAggregation,omitempty"`
+	State           int                   `json:"state"`
+	Lead            int                   `json:"lead,omitempty"`
+	Lag             int                   `json:"lag,omitempty"`
+}
+
+type Group struct {
```
Comment (smyrman) [inline fields/evaluate.go:158]:
```suggestion
type EvaluateGroup struct {
```

**Comment by smyrman** [comment]
Split in two documents, but try to follow this commit-style:

- https://searis.slite.com/app/docs/xvLCQUTVkH
- https://searis.slite.com/app/docs/qZbg75gJge

E.g. 
```
BAC-296: update Client.Evaluate call to version 1.2alpha1 (breaking)
<Blank line>
<Longer description, including list all breaking changes>
```

**Review by smyrman — APPROVED** [review]
